### PR TITLE
integration-tests: add network-bind interface test

### PR DESCRIPTION
**Pull request by fgimenez**
The test reuses the previous network-consumer snap as client and introduces a network-bind-consumer snap. It's been included in the autopkgtest suite.

### CODE REVIEW

File: integration-tests/data/snaps/network-bind-consumer/bin/consumer [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+import sys
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+class testHTTPServer_RequestHandler(BaseHTTPRequestHandler):
```
Comment (elopio) [inline integration-tests/data/snaps/network-bind-consumer/bin/consumer]: no need for an underscore in there.

File: integration-tests/tests/interfaces_test.go [hunk 2]
```diff
@@ -49,26 +49,32 @@ const (
...
 func (s *interfaceSuite) SetUpTest(c *check.C) {
 	s.SnappySuite.SetUpTest(c)
 
 	var err error
-	s.snapPath, err = build.LocalSnap(c, s.sampleSnap)
-	c.Assert(err, check.IsNil)
 
-	common.InstallSnap(c, s.snapPath)
+	s.snapPaths = make([]string, len(s.sampleSnaps))
+	for i := 0; i < len(s.sampleSnaps); i++ {
```
Comment (mvo5) [inline integration-tests/tests/interfaces_test.go]: If this is multiple snaps now, should  we switch to `SetUpTestSuite` (or what it is called in gocheck)?

Comment (mvo5) [inline integration-tests/tests/interfaces_test.go]: Nevermind, looking at this I think its fine.

**Comment by mvo5** [comment]
:+1:

File: integration-tests/data/snaps/network-bind-consumer/bin/consumer (hunk 1 above)
Comment (niemeyer) [inline integration-tests/data/snaps/network-bind-consumer/bin/consumer]: I'd just name it `testRequestHandler` instead.

File: integration-tests/tests/network_bind_interface_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	c.Assert(output, check.Equals, okOutput)
+
+	cli.ExecCommand(c, "sudo", "snap", "disconnect",
+		s.plug+":"+s.slot, "ubuntu-core:"+s.slot)
+
+	output = cli.ExecCommand(c, "snap", "interfaces")
+	c.Assert(output, check.Matches, disconnectedPattern(s.slot, s.plug))
+
+	output = cli.ExecCommand(c, "network-consumer", providerURL)
+	c.Assert(output, check.Not(check.Equals), okOutput)
```
Comment (niemeyer) [inline integration-tests/tests/network_bind_interface_test.go]: Can we be slightly more strict here about ensuring that the error we're getting reflects the problem we expect the snap to face when we disconnect the interface?  It's easy for the snap to be broken for reasons which do not reflect the real outcome we expect.

**Comment by niemeyer** [comment]
LGTM once you are happy about the points made.

**Comment by elopio** [comment]
:+1: with the note of the okOutput that we have just discussed.

File: integration-tests/data/snaps/network-bind-consumer/bin/consumer (hunk 1 above)
Comment (fgimenez) [inline integration-tests/data/snaps/network-bind-consumer/bin/consumer]: ok done

**Comment by fgimenez** [comment]
Thanks, i've also modified the networkInterface suite to reflect the same structure as this one, no dependencies on external service and exercising the interface against the service provided by a local snap.

**Comment by niemeyer** [comment]
Tests have failed.

**Comment by fgimenez** [comment]
@niemeyer yes, infra issues
retest this please

**Comment by fgimenez** [comment]
retest this please

**Comment by fgimenez** [comment]
all green, merging

**Pull request merged**